File: lib/models/order.dart
class Order {
  final String id;
  final String orderReference;
  final String customerName;
  final String street;
  final String city;
  final String postalCode;
  final String phone;
  final String deliveryOption;
  final double totalAmount;
  final double deliveryFee;
  final List<Map<String, dynamic>> items;
  final DateTime orderDate;
  final String status;

  Order({
    required this.id,
    required this.orderReference,
    required this.customerName,
    required this.street,
    required this.city,
    required this.postalCode,
    required this.phone,
    required this.deliveryOption,
    required this.totalAmount,
    required this.deliveryFee,
    required this.items,
    required this.orderDate,
    this.status = 'Pending',
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'orderReference': orderReference,
      'customerName': customerName,
      'street': street,
      'city': city,
      'postalCode': postalCode,
      'phone': phone,
      'deliveryOption': deliveryOption,
      'totalAmount': totalAmount,
      'deliveryFee': deliveryFee,
      'items': items,
      'orderDate': orderDate.toIso8601String(),
      'status': status,
    };
  }

  factory Order.fromMap(Map<String, dynamic> map) {
    return Order(
      id: map['id'],
      orderReference: map['orderReference'],
      customerName: map['customerName'],
      street: map['street'],
      city: map['city'],
      postalCode: map['postalCode'],
      phone: map['phone'],
      deliveryOption: map['deliveryOption'],
      totalAmount: map['totalAmount'].toDouble(),
      deliveryFee: map['deliveryFee'].toDouble(),
      items: List<Map<String, dynamic>>.from(map['items']),
      orderDate: DateTime.parse(map['orderDate']),
      status: map['status'],
    );
  }
} 

File: backend/models/Order.js
const mongoose = require('mongoose');

const orderSchema = new mongoose.Schema({
  orderReference: {
    type: String,
    required: true,
    unique: true
  },
  customerName: {
    type: String,
    required: true
  },
  street: {
    type: String,
    required: true
  },
  city: {
    type: String,
    required: true
  },
  postalCode: {
    type: String,
    required: true
  },
  phone: {
    type: String,
    required: true
  },
  deliveryOption: {
    type: String,
    required: true
  },
  totalAmount: {
    type: Number,
    required: true
  },
  deliveryFee: {
    type: Number,
    required: true
  },
  items: [{
    productId: {
      type: String,
      required: true
    },
    productName: {
      type: String,
      required: true
    },
    quantity: {
      type: Number,
      required: true
    },
    price: {
      type: Number,
      required: true
    },
    totalPrice: {
      type: Number,
      required: true
    }
  }],
  orderDate: {
    type: Date,
    default: Date.now
  },
  status: {
    type: String,
    enum: ['Pending', 'Confirmed', 'Processing', 'Shipped', 'Delivered', 'Cancelled'],
    default: 'Pending'
  },
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  }
}, {
  timestamps: true
});

// Generate order reference
orderSchema.pre('save', function(next) {
  if (!this.orderReference) {
    const timestamp = Date.now().toString().slice(-6);
    const random = Math.floor(Math.random() * 1000).toString().padStart(3, '0');
    this.orderReference = `ORD-${timestamp}-${random}`;
  }
  next();
});

module.exports = mongoose.model('Order', orderSchema); 